### PR TITLE
Fix state inconsistency in uniform-unstick

### PR DESCRIPTION
**Pull request by Jarkami**
fixes https://github.com/DFHack/dfhack/issues/5460

### CODE REVIEW

**Review by SilasD — COMMENTED** [review]
**This fixes the bug you're targeting**; it's fine.

I still think we should totally construct a copy of the data from scratch, and compare with the existing vector, but I know that's extra work for (probably) no gain.

I just see three possibilities for corruption:
* values that shouldn't be in this vector.  You're fixing that.
* values that *should* be in the vector but aren't.  I haven't probed for this, but it's probably not happening.
* the vector itself not being sorted.  I did check two forts with extensive military (in 0.47.05 and 0.51.*), and didn't find any cases of this.

I suppose this is just nitpicking.  So is my line-by-line; there's nothing that *needs* changed.

File: uniform-unstick.lua [hunk 1]
```diff
@@ -196,6 +196,22 @@ local function process(unit, args, need_newline)
...
+        local u_id = squad_position.equipment.assigned_items[i]
+        -- Quiver, backpack, and flask are assigned in their own locations rather than in equipment.uniform, and thus need their own checks
+        -- If more separately-assigned items are added in the future, this handling will need to be updated accordingly
+        if assigned_items[u_id] == nil and u_id ~= squad_position.equipment.quiver and u_id ~= squad_position.equipment.backpack and u_id ~= squad_position.equipment.flask then
+            local item = df.item.find(u_id)
+            if item ~= nil then
+                need_newline = print_line(unit_name .. " has an improperly assigned item, item # " .. u_id .. " '" .. item_description(item) .. "'; removing it")
+            else
+                need_newline = print_line(unit_name .. " has a nonexistent item assigned, item # " .. u_id .. "; removing it")
+            end
```
[the remark this answers is not in the document]
Comment (ab9rf) [reply uniform-unstick.lua]: This is entirely appropriate and should not be changed.

File: uniform-unstick.lua [hunk 2]
```diff
@@ -16,7 +16,7 @@ local validArgs = utils.invert({
 -- Functions
 
 local function item_description(item)
-    return dfhack.df2console(dfhack.items.getDescription(item, 0, true))
+    return "item #" .. item.id .. " '" .. dfhack.df2console(dfhack.items.getDescription(item, 0, true)) .. "'"
 end
```
[the remark this answers is not in the document]
Comment (ab9rf) [reply uniform-unstick.lua:20]: I did a cursory scan through the code and I think `item` will never be null on calls to this function, but I am not at all sure of that. If you're confident that this code will never be called with `item` as null, it's enough to document that `item` must never be null. Otherwise, add a test for item being null to this code and provide an appropriate alternative. Otherwise, the script will abort with an exception, which will generally be confusing to users (as most people never look at the console and thus won't see the exception).

**Comment by Jarkami** [comment]
Did a quick readability pass on the script as a whole. `print_line()`/`need_newline` is gone, some variable names were changed for clarity, plus a couple more small rearrangements where I felt things were extra confusing. Actual behavior should not be affected by any of these changes.

Ultimately with how many different structures this script is pulling from and comparing I'm not sure it's ever going to reach _fantastic_ readability, but hopefully this is at least a decent step in the right direction.
